scripts/config/webpack/util/env.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable prefer-import/prefer-import-over-require */
+
+const urlJoin = require('proper-url-join');
+const normalizeUrl = (url) => url ? urlJoin(url) : '';
+
+function getEnvVariables() {
+    // Construct the env variables, starting with `NODE_ENV`, `SITE_URL` and `PUBLIC_URL`.
+    // - `NODE_ENV` - Useful for determining whether we are running in production mode.
+    //   Most importantly, it switches React into the correct mode.
+    // - `PUBLIC_URL` - Useful when generating full URLs to files present in the `public` folder.
+    // - `SITE_URL` - Useful when generating full URLs to use in "copy to clipboard"
+    //   or in social share buttons.
+    const env = {
+        NODE_ENV: process.env.NODE_ENV || 'production',
+        PUBLIC_URL: normalizeUrl(process.env.PUBLIC_URL),
+        SITE_URL: normalizeUrl(process.env.SITE_URL),
+    };
+
+    // Add the `REACT_APP_*` into the output environment variables.
+    Object.keys(process.env)
+    .filter((key) => /REACT_APP_/.test(key))
+    .forEach((key) => {
+        env[key] = process.env[key];
+    });
+
+    return env;
+}
+
+// Stringify env variables so we can feed into Webpack DefinePlugin
+// Optionally you can also include `process.env` so that destructuring process.env works
+function inlineEnvVariables(envVars, options = {}) {
```

Comment:
Could you do `options = { includeProcessEnv, ...options }` instead? This ensures that the code is future-proof in case we later need non-boolean falsy options as defaults.